### PR TITLE
Remove white space from request URL

### LINKED ISSUE #11: Allow Query Compression to Prevent 413 Payload Too Large Error
Hey Prismic Team,

I'm currently getting an error for a query that works fine via the `my-repo.prismic.io/graphql` interface.

Here's a snippet of the error:

```javascript
 name: 'ServerParseError',
     response:
      Response {
        size: 0,
        timeout: 0,
        [Symbol(Body internals)]: [Object],
        [Symbol(Response internals)]: [Object] },
     statusCode: 413,
     bodyText:
      '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">\n<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>413 ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size="1px">\nBad request.\nWe can\'t connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.\n<BR clear="all">\nIf you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.\n<BR clear="all">\n<HR noshade size="1px">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: XVCInjS_BVP43_A4pIEdrowDhbW8XXokfyHIxHMnWRa7AoviPhIVpg==\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>' },
  message: 'Network error: Unexpected token < in JSON at position 0',
```
Specifically it's Cloudfront: `If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.`

Cloudfront is returning an HTML doc with the error which `apollo-client` obviously doesn't like since it expects JSON.

I've tried using a compression library: (https://github.com/jane/gql-compress) to reduce the size of the payload with no luck.

I wanted to put this out there in case anyone else is having this issue and see what possible solutions there are.

So far I can see a couple solves:

1. Make 2 queries and merge the responses(this is really not ideal).
2. Allow compression with `apollo-link-prismic`.

A user on the `prismic-club` slack channel proposed forking the repo and implementing something like this:

```javascript
const customFetch = (uri, options) => {
           const compressedURI = uri.replace(/(%0A|%20)*(%20|%7B|%7D)(%0A|%20)*/g, (m, p1, p2) => p2);
           return fetch(compressedURI, options);
       };

       const httpLink = new HttpLink({
           uri: `${baseURI}/graphql`,
           useGETForQueries: true,
           fetch: customFetch,
       }); 
```

Perhaps you can allow the final request to be passed a function that compresses the query?

Thoughts?

Thanks!

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
Trim excess white space characters from the request url.
Fix #11

### CODE REVIEW

File: tests/removeWhiteSpace.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        const output = "https://test.cdn.prismic.io/graphql?query=%7B_allDocuments%7Bedges%7Bnode%7B_meta%7Bid%7D%7D%7D%7D%7D&variables=%7B%7D"
+        const result = removeWhiteSpace(input);
+        expect(result).toBe(output);
+    });
+
+    it("no match", () => {
+        const input = "https://www.example.com/";
+        const output = input;
+        const result = removeWhiteSpace(input);
+        expect(result).toBe(input);
```
Comment (ahafidi) [inline tests/removeWhiteSpace.test.js:54]: Remove the useless output var or use it instead of input

File: lib/removeWhiteSpace.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+function removeWhiteSpace(str) {
```
Comment (ahafidi) [inline lib/removeWhiteSpace.js]: This function is not bundled correctly with babel, so either you move this function in the `index.js` file or you add something like `rollup` to this project.

Comment (Raph22) [reply lib/removeWhiteSpace.js]: Or use CommonJS module. TBD what is the best solution

**Review by Raph22 — CHANGES_REQUESTED** [review]

**Review by ahafidi — CHANGES_REQUESTED** [review]
See the comments left in your code changes

**Review by ahafidi — APPROVED** [review]

**Review by ahafidi — APPROVED** [review]

**Review by Raph22 — APPROVED** [review]

**Pull request merged**